lib/Doctrine/Common/DataFixtures/ReferenceRepository.php

```diff
@@ -158,8 +162,8 @@ public function getReference($name)
 
         $reference = $this->references[$name];
         $meta = $this->manager->getClassMetadata(get_class($reference));
-        $uow = $this->manager->getUnitOfWork();
-        if (!$uow->isInIdentityMap($reference) && isset($this->identities[$name])) {
+
+        if (!$this->hasIdentifier($reference) && isset($this->identities[$name])) {
```

Comment:
I think this case could actually use ``$this->manager->contains()`` as we want to determine whether the object is managed